### PR TITLE
Add support for codeowners to repo

### PR DESCRIPTION
**Pull request by ryanbogan**
Signed-off-by: Ryan Bogan <rbogan@amazon.com>

### Description
Add support for codeowners to repo
 
### Issues Resolved
Progress on:
[Meta issue](https://github.com/opensearch-project/project-meta/issues/31)
 
### Check List
- [ X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by dlvenable — APPROVED** [review]

**Review by graytaylor0 — APPROVED** [review]

**Comment by CEHENKLE** [comment]
@opensearch-project/data-prepper I've updated the branch rules to use this codeowner file.  Do you want the number of PRs required set to 1 or 2?

**Comment by dlvenable** [comment]
We would like to set the required number of approvals to 1. Thank you!

**Comment by CEHENKLE** [comment]
I've updated the branch protection rules.  Thanks.

**Pull request merged**
